rootfs-builder/README.md

```diff
@@ -0,0 +1,57 @@
+# Building a rootfs for Kata Containers Guest OS #
+
+Kata Containers guest OS is created using `rootfs.sh`.
+
+## Supported base OS ##
+
+The `rootfs.sh` script support different based OS `rootfs`. To build
+a `rootfs` based in a supported based OS run:
+
+``` ./rootfs.sh <distro> ```
+
+To check the supported `rootfs` based OS use:
+
+```
+./rootfs.sh -h
+```
+
+## Adding support for new base OS ##
+
+The script `rootfs.sh` will look for directories that meets the following
+structure:
+
+- A bash script that called `rootfs_lib.sh`
+
+This file must contain a function called `build_rootfs()` this function
+must receive as first argument the path where the `rootfs` will be
+populated.
+
+
+- A bash file `config.sh`
+
+This represents the specific configuration for each based OS. It must
+provide configuration specific variables for users modify them as needed.
+The config file will be loaded before execute `build_rootfs()` to provide
+all the need configuration to the function.
+
+```bash
+#Arguments
+#rootfs_dir=$1
+#Optional environment variables
+#REPO_URL: URL to distribution repository ( should be configured in 
+#          config.sh file)
+#EXTRA_PKGS: Variable to add extra PKGS provided by the user
+#Any other configuration variable for a specific distro must be added 
+#and documented on its own config.sh
+#
+#Expected result
+#rootfs_dir populated with rootfs pkgs
+#It must provide a binary in /sbin/init
+build_rootfs() {
```

Comment:
This is really helpful, but I suggest that rather than embedding this into the document you create something like `rootfs_lib_template.sh` and change this section to say that you should:

- Copy `rootfs_lib_template.sh` to a particular directory.
- Implement the stub `build_rootfs()` function.

Reply:
@jodh-intel  added to `example` directory and added a make file to create a configuration based in templates as well as documented, please take a look.